### PR TITLE
Added code to capture flask app metrics

### PR DESCRIPTION
**Pull request by akkuldn**
Added integration of Prometheus flask exporter to fetch various metrics from the newsletter application.
The metrics should be visible at `/metrics` endpoint
**To test:**
- Install the requirements
- Run the newsletter application on your local system
- Visit any webpage on the newsletter server
- Go to the metrics page, and search for "flask_http_request_duration_seconds_count"
-  The count for this metric should increment each time you visit the page. For example if you visit `/articles` page then the metric ```flask_http_request_duration_seconds_count{method="GET",path="/articles",status="200"}``` would increment each time you visit the page.

### CODE REVIEW

File: newsletter_automation/newsletter/routes.py [hunk 1]
```diff
@@ -88,12 +89,13 @@ def logout():
         app.logger.error(e)
     return redirect('/')
 
-
 @app.route('/home')
 @app.route('/')
 @Authentication_Required.requires_auth
 def index():
-    return render_template('home.html', title="Home")
+    time.sleep(6)
```
Comment (qxf2) [inline newsletter_automation/newsletter/routes.py:96]: Why is there a sleep here I am guessing you forgot to delete the line after debugging?

File: newsletter_automation/newsletter/routes.py [hunk 2]
```diff
@@ -21,6 +22,8 @@
 from prometheus_client import generate_latest, CONTENT_TYPE_LATEST
 from prometheus_client import Counter, Gauge
 import psutil
+import time
```
Comment (qxf2) [inline newsletter_automation/newsletter/routes.py:25]: Do we need these imports?

File: newsletter_automation/newsletter/routes.py [hunk 3]
```diff
@@ -21,6 +22,8 @@
...
+from functools import wraps
```
Comment (qxf2) [inline newsletter_automation/newsletter/routes.py:26]: Where are we using wraps?

File: newsletter_automation/newsletter/routes.py [hunk 4]
```diff
@@ -512,3 +515,4 @@ def prometheus_metrics():
     cpu_usage.set(cpu_percent)
```
Comment (qxf2) [inline newsletter_automation/newsletter/routes.py:515]: Are you overwriting the prometheus metric? I did not follow the change to this method. I was assuming you would not touch this method except to say do not track this method.

**Review by qxf2 — CHANGES_REQUESTED** [review]
I am done reviewing your PR @akkuldn . Please take a look.

File: newsletter_automation/newsletter/routes.py (hunk 1 above)
Comment (akkuldn) [reply newsletter_automation/newsletter/routes.py]: Yes I forgot to remove it after debugging. Removed it now

File: newsletter_automation/newsletter/routes.py (hunk 2 above)
Comment (akkuldn) [reply newsletter_automation/newsletter/routes.py]: Removed the time and wraps import. The rest is necessary

File: newsletter_automation/newsletter/routes.py (hunk 3 above)
Comment (akkuldn) [reply newsletter_automation/newsletter/routes.py]: I had added it as a part of the first implementation approach, forgot to remove the import. Removed now

File: newsletter_automation/newsletter/routes.py (hunk 4 above)
Comment (akkuldn) [reply newsletter_automation/newsletter/routes.py:511]: I am not touching this method. I'm just telling flask exporter to ignore tracking this endpoint. No other changes

File: newsletter_automation/newsletter/routes.py [hunk 5]
```diff
@@ -126,6 +124,7 @@ def add_articles():
 @Authentication_Required.requires_auth
 def articles():
     """To add articles through pages"""
+    #app.logger.info(request.path)
```
Comment (qxf2) [inline newsletter_automation/newsletter/routes.py:127]: Please delete this commented out line of code. In general, there should be no commented out lines of code.

**Review by qxf2 — CHANGES_REQUESTED** [review]
@akkuldn - tiny review comment. You can resolve it and merge to main.

File: newsletter_automation/newsletter/routes.py (hunk 5 above)
Comment (akkuldn) [reply newsletter_automation/newsletter/routes.py]: I have removed all the commented out codes from routes file

**Pull request merged**